src/v/raft/heartbeat_manager.h

```diff
@@ -144,7 +144,7 @@ class heartbeat_manager {
     /// \param result if the node return successful heartbeats
     void process_reply(
       model::node_id n,
-      absl::flat_hash_map<raft::group_id, follower_request_meta> groups,
+      absl::node_hash_map<raft::group_id, follower_request_meta> groups,
```

Comment:
would this be all the raft groups on a particular node? at 16k raft groups this would be at a minimum 128kb contiguous allocation. I wonder it makes sense to move directly to something like std::map or abs::btree_map which are slower but naturally fragment memory? I'm sort of speaking generally about all the places you think might grow, not just this one case.

Reply:
i was considering that, i will change this to `absl::btree_map` for now.